fdbrpc/FlowTransport.actor.cpp

```diff
@@ -1325,14 +1376,19 @@ static ReliablePacket* sendPacket(TransportData* self, Reference<Peer> peer, ISe
 	int prevBytesWritten = pb->bytes_written;
 	PacketBuffer* checksumPb = pb;
 
-	PacketWriter wr(pb,rp,AssumeVersion(currentProtocolVersion));  // SOMEDAY: Can we downgrade to talk to older peers?
+	PacketWriter wr(pb,rp,AssumeVersion(g_network->protocolVersion()));  // SOMEDAY: Can we downgrade to talk to older peers?
 
 	// Reserve some space for packet length and checksum, write them after serializing data
 	SplitBuffer packetInfoBuffer;
-	uint32_t len, checksum = 0;
+
+	auto protocolChecksumIter = protocolToChecksum.upper_bound(g_network->protocolVersion());
```

Comment:
We need to use most recent checksum supported by both us and the peer. We'll need to wait for the Peer's ConnectPacket before we can send now